numpy/core/overrides.py

```diff
@@ -44,43 +55,75 @@ def get_overloaded_types_and_args(relevant_args):
     return overloaded_types, overloaded_args
 
 
-def array_function_override(overloaded_args, func, types, args, kwargs):
-    """Call __array_function__ implementations."""
+def array_function_implementation_or_override(
+        implementation_func, api_func, dispatcher, args, kwargs):
```

Comment:
Might it make more sense to call the first one `ndarray_implementation`? That is what it would be if we actually worked via `ndarray.__array_ufunc__`. A nice aspect of that is that one could rename `api_func` to `function`.

Or perhaps just `implementation` and `function`?

Reply:
How about `implementation` and `public_api`?